src/pymor/algorithms/error.py

```diff
@@ -257,56 +264,121 @@ def custom_value(rom, fom, reductor, mu, dim):
-            legend = []
-            if error_norms:
-                for name, errors in zip(error_norm_names, max_errors):
-                    ax.semilogy(basis_sizes, errors)
-                    legend.append(name)
-            if error_estimator:
-                ax.semilogy(basis_sizes, max_error_estimates)
-                legend.append('error estimator')
-            ax.legend(legend)
-            ax.set_title('maximum errors')
-            current_plot += 1
-
-        if bool(error_norms) and error_estimator:
-            ax = fig.add_subplot(1, num_plots, current_plot)
-            ax.semilogy(basis_sizes, min_effectivities)
-            ax.semilogy(basis_sizes, max_effectivities)
-            ax.legend(('min', 'max'))
-            ax.set_title('error estimator effectivities')
-            current_plot += 1
+        result['figure'] = plot_reduction_error_analysis(
+                result, None, plot_custom_logarithmic=plot_custom_logarithmic, return_fig=True)
 
-        if condition:
-            ax = fig.add_subplot(1, num_plots, current_plot)
-            ax.semilogy(basis_sizes, max_conditions)
-            ax.set_title('maximum condition')
-            current_plot += 1
-
-        if custom:
-            ax = fig.add_subplot(1, num_plots, current_plot)
-            legend = []
-            for i, values in enumerate(custom_values):
-                if plot_custom_logarithmic:
-                    ax.semilogy(basis_sizes, values)
-                else:
-                    ax.plot(basis_sizes, values)
-                legend.append('value ' + str(i))
-            ax.legend(legend)
-            ax.set_title('maximum custom values')
-            current_plot += 1
+    return result
 
-        result['figure'] = fig
 
-    return result
+
+def plot_reduction_error_analysis(result, max_basis_size=None, plot_effectivities=True, plot_condition=True,
+        plot_custom_logarithmic=True, plot_custom_with_errors=False, return_fig=False):
+
+    error_norms = 'norms' in result
+    error_estimator = 'error_estimates' in result
+    condition = 'conditions' in result
+    custom = 'custom_values' in result
+
+    basis_sizes = result['basis_sizes']
+    if error_norms:
+        error_norm_names = result['error_norm_names']
+        max_errors = result['max_errors']
+        errors = result['errors']
+    if error_estimator:
+        max_estimates = result['max_error_estimates']
+    if error_norms and error_estimator:
+        min_effectivities = result['min_effectivities']
+        max_effectivities = result['max_effectivities']
+    if condition:
+        max_conditions = result['max_conditions']
+    if custom:
+        custom_values = result['custom_values']
+        custom_names = result['custom_names']
+
+    max_basis_size = max_basis_size if max_basis_size else len(basis_sizes)
+
+    import matplotlib.pyplot as plt
+    from matplotlib.colors import TABLEAU_COLORS as COLORS
+    colors = [[]]
```

Comment:
Couldn't this just be a flat list?

Reply:
```python
UnboundLocalError: local variable 'colors' referenced before assignment
```

Reply:
Any thoughts here? Would making `colors` a plain list and declaring it global in `get_color` be more elegant?

Reply:
```python
from itertoos import cycle
color = cycle(reversed(COLORS.keys()))
...
next(color)
```
should do the trick.

Reply:
Ah, didn't recognize you used the list nest to get a modifiable captured var at first. You could use nonlocal here, but really what you want is to use `colors = itertools.cycle(COLORS.keys())` and then `next(colors)` instead of the function call.

Reply:
ah, thx